### PR TITLE
removes hop-by-hop headers in responses

### PR DESCRIPTION
**Pull request by shamsimam**
## Changes proposed in this PR

- removes connection-specific header fields when transmitting HTTP/1.x message via HTTP/2

## Why are we making these changes?

> HTTP/2 does not use the Connection header field to indicate connection-specific header fields; in this protocol, connection-specific metadata is conveyed by other means. An endpoint MUST NOT generate an HTTP/2 message containing connection-specific header fields; any message containing connection-specific header fields MUST be treated as malformed (Section 8.1.2.6).
> 
> The only exception to this is the TE header field, which MAY be present in an HTTP/2 request; when it is, it MUST NOT contain any value other than "trailers".
> 
> This means that an intermediary transforming an HTTP/1.x message to HTTP/2 will need to remove any header fields nominated by the Connection header field, along with the Connection header field itself. Such intermediaries SHOULD also remove other connection-specific header fields, such as Keep-Alive, Proxy-Connection, Transfer-Encoding, and Upgrade, even if they are not nominated by the Connection header field.

https://http2.github.io/http2-spec/

### CODE REVIEW

File: waiter/src/waiter/process_request.clj [hunk 1]
```diff
@@ -408,13 +409,23 @@
   (track-process-error-metrics descriptor)
   (utils/exception->response exception request))
 
+(defn cleanup-response-headers
+  "Removes connection-specific header fields when transmitting HTTP/1.x message via HTTP/2."
+  [{:keys [headers] :as response} internal-protocol backend-protocol]
+  (cond-> response
+    (and (= internal-protocol "HTTP/2.0") (not= backend-protocol "HTTP/2.0"))
```
Comment (sradack) [inline waiter/src/waiter/process_request.clj]: Shouldn't we be removing these headers in all cases, not use HTTP/2?  See https://www.mnot.net/blog/2011/07/11/what_proxies_must_do "Remove Hop-by-hop headers"

Comment (sradack) [reply waiter/src/waiter/process_request.clj]: We do this already for the request headers.

Comment (shamsimam) [reply waiter/src/waiter/process_request.clj]: I had missed that hop-by-hop headers need to be removed for responses too. Thanks, done.

File: waiter/src/waiter/process_request.clj [hunk 2]
```diff
@@ -23,6 +23,7 @@
             [metrics.histograms :as histograms]
             [metrics.meters :as meters]
             [metrics.timers :as timers]
+            [plumbing.core :as pc]
```
Comment (sradack) [inline waiter/src/waiter/process_request.clj]: unused

File: waiter/src/waiter/process_request.clj [hunk 3]
```diff
@@ -414,7 +415,7 @@
     [make-request-fn instance-rpc-chan start-new-service-fn
      instance-request-properties determine-priority-fn process-backend-response-fn
      request-abort-callback-factory local-usage-agent
-     {:keys [ctrl descriptor request-id request-time] :as request}]
+     {:keys [ctrl descriptor internal-protocol request-id request-time] :as request}]
```
Comment (sradack) [inline waiter/src/waiter/process_request.clj]: unused

Comment (shamsimam) [reply waiter/src/waiter/process_request.clj]: Removed.

File: waiter/src/waiter/process_request.clj (hunk 2 above)
Comment (shamsimam) [reply waiter/src/waiter/process_request.clj]: Removed.

**Comment by shamsimam** [comment]
@sradack ready to merge (unrelated CLI tests failing).

**Pull request merged**